massa-grpc/src/stream/new_slot_execution_outputs.rs

```diff
@@ -400,14 +439,170 @@ fn filter_map_exec_output(
+                            if let Some(change_types) = &async_pool_changes_filter.change_types {
+                                if !change_types
+                                    .contains(&(grpc_model::AsyncPoolChangeType::Set as i32))
+                                {
+                                    return false;
+                                }
+                            }
+                            //TODO to be confirmed
+                            if let Some(handlers) = &async_pool_changes_filter.handlers {
+                                if !handlers.contains(&value.function) {
+                                    return false;
+                                }
+                            }
+                            if let Some(destination_addresses) =
+                                &async_pool_changes_filter.destination_addresses
+                            {
+                                if !destination_addresses.contains(&value.destination) {
+                                    return false;
+                                }
+                            }
+
+                            if let Some(emitter_addresses) =
+                                &async_pool_changes_filter.emitter_addresses
+                            {
+                                if !emitter_addresses.contains(&value.sender) {
+                                    return false;
+                                }
+                            }
+
+                            if let Some(can_be_executed) = async_pool_changes_filter.can_be_executed
+                            {
+                                if value.can_be_executed != can_be_executed {
+                                    return false;
+                                }
+                            }
+                        }
+                        SetUpdateOrDelete::Update(value) => {
+                            if let Some(change_types) = &async_pool_changes_filter.change_types {
+                                if !change_types
+                                    .contains(&(grpc_model::AsyncPoolChangeType::Set as i32))
+                                {
+                                    return false;
+                                }
+                            }
+                            //TODO to be confirmed
+                            if let Some(handlers) = &async_pool_changes_filter.handlers {
+                                match value.function.clone() {
+                                    massa_ledger_exports::SetOrKeep::Set(function_name) => {
+                                        if !handlers.contains(&function_name) {
+                                            return false;
+                                        }
+                                    }
+                                    //TODO to be confirmed
+                                    massa_ledger_exports::SetOrKeep::Keep => {
```

Comment:
@AurelienFT if It's a Keep, we can't compare to anything, should we do something or act like there is nothing ?

Reply:
If a change is here with a `function` set to `Keep` then there is at least another change that is here with a `Set` and so the choice of returning it or not should depends on the choice on the field that have the `Set`

Reply:
We'll take a decision when this PR will be prioritised.